code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/contentcards/ContentCardSchemaDataUtils.kt

```diff
@@ -0,0 +1,177 @@
+*/
+
+package com.adobe.marketing.mobile.messaging.contentcards
+
+import com.adobe.marketing.mobile.aepcomposeui.AepUI
+import com.adobe.marketing.mobile.aepcomposeui.SmallImageUI
+import com.adobe.marketing.mobile.aepcomposeui.state.SmallImageCardUIState
+import com.adobe.marketing.mobile.aepcomposeui.uimodels.AepButton
+import com.adobe.marketing.mobile.aepcomposeui.uimodels.AepDismissButton
+import com.adobe.marketing.mobile.aepcomposeui.uimodels.AepImage
+import com.adobe.marketing.mobile.aepcomposeui.uimodels.AepText
+import com.adobe.marketing.mobile.aepcomposeui.uimodels.AepUITemplate
+import com.adobe.marketing.mobile.aepcomposeui.uimodels.SmallImageTemplate
+import com.adobe.marketing.mobile.messaging.ContentCardSchemaData
+import com.adobe.marketing.mobile.messaging.MessagingConstants
+import com.adobe.marketing.mobile.messaging.Proposition
+import com.adobe.marketing.mobile.messaging.SchemaType
+import com.adobe.marketing.mobile.util.DataReader
+
+/**
+ * Utility object for creating instances of AEP model classes from Map data structures.
+ * This utility handles parsing and transforming content card data into the relevant AEP UI models,
+ * such as [AepText], [AepImage], [AepButton], and [AepDismissButton], as well as creating
+ * the corresponding UI templates like [SmallImageTemplate].
+ */
+internal object ContentCardSchemaDataUtils {
+    /**
+     * Creates an instance of [AepText] from a map of key-value pairs.
+     *
+     * @param map The map containing the data for [AepText].
+     * @return An instance of [AepText].
+     */
+    private fun createAepText(map: Map<String, Any>): AepText {
+        return AepText(
+            content = DataReader.optString(map, MessagingConstants.ContentCard.UIKeys.CONTENT, "")
+        )
+    }
+
+    /**
+     * Creates an instance of [AepImage] from a map of key-value pairs.
+     *
+     * @param map The map containing the data for [AepImage].
+     * @return An instance of [AepImage].
+     */
+    private fun createAepImage(map: Map<String, Any>): AepImage {
+        return AepImage(
+            url = DataReader.optString(map, MessagingConstants.ContentCard.UIKeys.URL, null),
+            darkUrl = DataReader.optString(map, MessagingConstants.ContentCard.UIKeys.DARK_URL, null)
+        )
+    }
+
+    /**
+     * Creates an instance of [AepButton] from a map of key-value pairs.
+     *
+     * @param map The map containing the data for [AepButton].
+     * @return An instance of [AepButton].
+     */
+    private fun createAepButton(map: Map<String, Any>): AepButton {
+        return AepButton(
+            id = DataReader.optString(map, MessagingConstants.ContentCard.UIKeys.INTERACT_ID, null),
+            actionUrl = DataReader.optString(map, MessagingConstants.ContentCard.UIKeys.ACTION_URL, null),
+            text = DataReader.optTypedMap(Any::class.java, map, MessagingConstants.ContentCard.UIKeys.TEXT, null).let { createAepText(it) }
```

Comment:
id, actionUrl and text for a `AepButton` cannot be null. This will lead to runtime exception. You might need to check they are non null before creating `AepButton`